src/server/helpers/utils.js

```diff
@@ -90,71 +60,63 @@ function getRevisionModels(orm) {
 		WorkRevision
 	];
 }
-async function getCompleteRevision(revision, orm) {
-	const {Entity, AliasSet} = orm;
+async function 	getCompleteRevision(revision, orm) {
```

Comment:
So this was starting to work quite well, but with 100 results per page loading the next page was quite slow.
Looking at the way we're constructing the complete revision, we do a lot of requests:
5 (one for each entity type) * number of revisions * 1 or 2 (getEntityParentAlias might be called)

Instead, we can fetch once for each entity type all the entity_revisions that match all our revisions, in one go, and then for each fetch the information we need. As a quick note, even though fetching the Entity model is much cleaner than my table name parsing hack, it does add a request per entity, so it might be worth comparing the speed at which the page loads with both configurations, if we want to further optimize.

Here's what I ended up with:
```
/* eslint-disable no-await-in-loop */
/**
 * Fetches the entities affected by a revision, their alias
 * or in case of deleted entities their last know alias.
 * It then attaches the necessary information to each revisions's entities array
 *
 * @param {array} revisions - the array of revisions
 * @param {object} orm - the BookBrainz ORM, initialized during app setup
 * @returns {array} - The modified revisions array
 */
async function getCompleteRevision(revisions, orm) {
	const revisionIDs = revisions.map(({revisionId}) => revisionId);
	const RevisionModels = getRevisionModels(orm);
	const {Entity} = orm;
	for (let i = 0; i < RevisionModels.length; i++) {
		const EntityRevision = RevisionModels[i];
		const entityRevisions = await new EntityRevision()
			.query((qb) => {
				qb.whereIn('id', revisionIDs);
			})
			.fetchAll({
				require: false,
				withRelated: [
					'data.aliasSet.defaultAlias'
				]
			})
			.catch(EntityRevision.NotFoundError, (err) => {
				// eslint-disable-next-line no-console
				console.log(err);
			});
		if (entityRevisions && entityRevisions.length) {
			const entityRevisionsJSON = entityRevisions.toJSON();
			for (let index = 0; index < entityRevisionsJSON.length; index++) {
				const entityRevision = entityRevisionsJSON[index];
				const entity = await new Entity({bbid: entityRevision.bbid}).fetch();
				const type = entity.get('type');
				const bbid = entity.get('bbid');
				const entityProps = {bbid, type};
				if (entityRevision.data) {
					entityProps.defaultAlias = entityRevision.data.aliasSet.defaultAlias;
				}
				// Fetch the parent alias only if data property is nullish, i.e. it is deleted
				else {
					entityProps.parentAlias = await orm.func.entity.getEntityParentAlias(orm, type, bbid);
				}
				// Find the revision by id and attach the current entity to it
				const revisionIndex = revisions.findIndex(rev => rev.revisionId === entityRevision.id);
				revisions[revisionIndex].entities.push(entityProps);
			}
		}
	}
	return revisions;
}
/* eslint-enable no-await-in-loop */
```